client/components/data/query-post-counts/index.jsx

```diff
@@ -1,58 +1,35 @@
-/** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestPostCounts } from 'state/posts/counts/actions';
 import { isRequestingPostCounts } from 'state/posts/counts/selectors';
 
-class QueryPostCounts extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId === nextProps.siteId && this.props.type === nextProps.type ) {
-			return;
-		}
-
-		this.request( nextProps );
-	}
+export default function QuerySiteDomains( { siteId, type } ) {
+	const requesting = useSelector( state => isRequestingPostCounts( state, siteId, type ) );
```

Comment:
This is an interesting one to consider, too. Does this look like the right approach for using selectors with parameters?

Reply:
Update, from elsewhere: it's possible to drop these altogether by moving the checking to a thunk. Did so.

Reply:
I think this works perfectly and has no downsides. The inline selector will be called on each Redux store update, with the latest and correct `siteId` and `type`, and will trigger component rerender only if the resulting value changes (i.e., not `===` to the previous one).

The inline function doesn't need to be memoized, there's no place where it would be compared with the previous value. Only the result it returns counts.